rclpy/rclpy/node.py

```diff
@@ -217,12 +231,109 @@ def get_logger(self):
         """Get the nodes logger."""
         return self._logger
 
+    def declare_parameter(
+        self,
+        name: str,
+        value: ParameterValue,
+        descriptor: ParameterDescriptor
+    ) -> Parameter:
+        """
+        Declare and initialize a parameter.
+
+        This method, if successful, will result in any callback registered with
+        :func:`set_parameters_callback` to be called.
+
+        :param name: Name of the parameter to declare.
+        :param value: Value of the parameter to declare.
+        :param descriptor: Descriptor for the parameter to declare.
+        :return: Parameter with the effectively assigned value.
+        :raises: ParameterAlreadyDeclaredException if the parameter had already been declared.
+        :raises: InvalidParameterException if the parameter name is invalid.
+        :raises: InvalidParameterValueException if the registered callback rejects the parameter.
+        """
+        return self.declare_parameters('', [(name, value, descriptor)])[0]
+
+    def declare_parameters(
+        self,
+        namespace: str,
+        parameters: List[Tuple[str, ParameterValue, ParameterDescriptor]]
```

Comment:
Similar to my previous comment, the value and descriptor could be optional:
```suggestion
        parameters: List[Tuple[str, Optional[ParameterValue], Optional[ParameterDescriptor]]]
```

Reply:
Addressed in https://github.com/ros2/rclpy/pull/325/commits/0925fe4e4a72a4b32ff1ec58790d73c4ba35838a.